protocol/jsonlines/decoder.go

```diff
@@ -40,17 +40,41 @@ func NewDecoder(r io.Reader) Decoder {
 	return &decoder{r: lr}
 }
 
+func (d *decoder) discardPending() error {
+	// Read and discard everytihng until the next newline
```

Comment:
This line may be better above the function declaration, to document the function